sui_types/src/messages.rs

```diff
@@ -575,15 +575,15 @@ impl SignedTransaction {
 impl Hash for SignedTransaction {
     fn hash<H: Hasher>(&self, state: &mut H) {
         self.data.hash(state);
-        self.auth_signature.authority.hash(state);
+        self.auth_signature.hash(state);
     }
 }
 
 impl PartialEq for SignedTransaction {
     fn eq(&self, other: &Self) -> bool {
-        // We do not compare the signatures, because there can be multiple
+        // We do not compare the tx_signature, because there can be multiple
         // valid signatures for the same data and signer.
-        self.data == other.data && self.auth_signature.authority == other.auth_signature.authority
+        self.data == other.data && self.auth_signature == other.auth_signature
```

Comment:
Ideally we should take the opportunity in a future PR to rename the `TransactionEnvelope`s field `auth_signature` to `auth_sign_info`, because it's very confusing when reading this; if we compare signature bytes or not (and this case we don't).

Reply:
Done